holidays/calendars/sinhala.py

```diff
@@ -0,0 +1,414 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from collections.abc import Iterable
+from datetime import date
+from typing import Optional
+
+from holidays.calendars.custom import _CustomCalendar
+from holidays.calendars.gregorian import JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.helpers import _normalize_tuple
+
+DURUTHU_POYA = "DURUTHU_POYA"
+NAWAM_POYA = "NAWAM_POYA"
+MEDIN_POYA = "MEDIN_POYA"
+BAK_POYA = "BAK_POYA"
+VESAK_POYA = "VESAK_POYA"
+POSON_POYA = "POSON_POYA"
+ESALA_POYA = "ESALA_POYA"
+NIKINI_POYA = "NIKINI_POYA"
+BINARA_POYA = "BINARA_POYA"
+VAP_POYA = "VAP_POYA"
+IL_POYA = "IL_POYA"
+UNDUVAP_POYA = "UNDUVAP_POYA"
+
+
+class _SinhalaLunar:
+    """
+    Sinhala Lunar calendar for 2003-2025 years.
+
+    Their Buddhist Uposatha day calculation method is different from Thai LuniSolar
+    and Buddhist (Mahayana) used in East Asia.
+
+    Due to the fact that Poya (Uposatha) days are calculated astronomically
+    based on how close a particular day is closest to full moon at noon, and that
+    an extra month is added every 33 months interval, this is hardcoded for now.
+
+    Adhi month dates are instead hardcoded in Sri Lanka country implementation.
+    """
+
+    START_YEAR = 2003
+    END_YEAR = 2025
+
+    DURUTHU_POYA_DATES = {
+        2003: (JAN, 17),
+        2004: (JAN, 7),
+        2005: (JAN, 24),
+        2006: (JAN, 13),
+        2007: (JAN, 3),
+        2008: (JAN, 22),
+        2009: ((JAN, 10), (DEC, 31)),
+        2011: (JAN, 19),
+        2012: (JAN, 8),
+        2013: (JAN, 26),
+        2014: (JAN, 15),
+        2015: (JAN, 4),
+        2016: (JAN, 23),
+        2017: (JAN, 12),
+        2018: (JAN, 1),
+        2019: (JAN, 20),
+        2020: (JAN, 10),
+        2021: (JAN, 28),
+        2022: (JAN, 17),
+        2023: (JAN, 6),
+        2024: (JAN, 25),
+        2025: (JAN, 13),
+    }
+
+    NAWAM_POYA_DATES = {
+        2003: (FEB, 16),
+        2004: (FEB, 5),
+        2005: (FEB, 23),
+        2006: (FEB, 12),
+        2007: (FEB, 1),
+        2008: (FEB, 20),
+        2009: (FEB, 9),
+        2010: (JAN, 29),
+        2011: (FEB, 17),
+        2012: (FEB, 7),
+        2013: (FEB, 25),
+        2014: (FEB, 14),
+        2015: (FEB, 3),
+        2016: (FEB, 22),
+        2017: (FEB, 10),
+        2018: (JAN, 31),
+        2019: (FEB, 19),
+        2020: (FEB, 8),
+        2021: (FEB, 26),
+        2022: (FEB, 16),
+        2023: (FEB, 5),
+        2024: (FEB, 23),
+        2025: (FEB, 13),
```

Comment:
```suggestion
        2025: (FEB, 12),
```
According to [https://sl.statebank/holiday-calendar](SBI).

Reply:
Nice catch ❤️